File: Swift/Leetcode.playground/Sources/2_String/LC_2609_FindTheLongestBalancedSubstringOfABinaryString.swift
class Solution {
    func findTheLongestBalancedSubstring(_ s: String) -> Int {
        let n = s.count
        var idx = 0
        var ans = 0
        while idx < n {
            var a = 0
            var b = 0
            while idx < n && s[s.index(s.startIndex, offsetBy: idx)] == "0" {
                a += 1
                idx += 1
            }
            while idx < n && s[s.index(s.startIndex, offsetBy: idx)] == "1" {
                b += 1
                idx += 1
            }
            ans = max(ans, min(a, b) * 2)
        }
        return ans
    }
}

File: Java/2_String/LC_2609_FindTheLongestBalancedSubstringOfABinaryString.java
class Solution {
    public int findTheLongestBalancedSubstring(String s) {
        int n = s.length(), idx = 0, ans = 0;
        while (idx < n) {
            int a = 0, b = 0;
            while (idx < n && s.charAt(idx) == '0') {
                a++;
                idx++;
            }
            while (idx < n && s.charAt(idx) == '1') {
                b++;
                idx++;
            }
            ans = Math.max(ans, Math.min(a, b) * 2);
        }
        return ans;
    }
}

public class LC_2609_FindTheLongestBalancedSubstringOfABinaryString {

}
